pkg/controller/plan/kubevirt.go

```diff
@@ -2686,21 +2690,40 @@ func (r *VirtualMachine) Conditions() (cnd *libcnd.Conditions) {
 	return
 }
 
-// Create an OwnerReference from a VM.
-func vmOwnerReference(vm *cnv.VirtualMachine) (ref meta.OwnerReference) {
-	blockOwnerDeletion := true
+// Generic function to create OwnerReferences based on the input type.
+func createOwnerReferences[T any](obj T) []meta.OwnerReference {
```

Comment:
Would prefer not to use any generics.
To make it general we can use the metadata which all k8s resources have.
```
func createOwnerReferences(objMeta metav1.ObjectMeta, typeMeta metav1.TypeMeta) []meta.OwnerReference {
	blockOwnerDeletion := false
	isController := false
	return []meta.OwnerReference{
		{
			APIVersion:         typeMeta.APIVersion,
			Kind:               typeMeta.Kind,
			Name:               objMeta.Name,
			UID:                objMeta.UID,
			Controller:         &isController,
			BlockOwnerDeletion: &blockOwnerDeletion,
		},
	}
}
```

usage:
```
createOwnerReferences(virtualMachine.ObjectMeta, virtualMachine.TypeMeta)
```